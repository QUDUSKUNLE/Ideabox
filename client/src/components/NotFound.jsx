import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @function NotFound
 * @description the NotFound component
 * @returns {string} The HTML markup for the NotFound component
 */
const NotFound = () =>
  (
    <div className="container">
      <div className="logo">
        <h4>IdeaBox</h4>
      </div>
      <div className="col s12 flex-center">
        <div className="center-align">
          <div className="center-align centralize">
            <h3 className="white-text">
              You`re lost.<Link to="/">Home</Link>
            </h3>
          </div>
        </div>
      </div>
    </div>
  );

export default NotFound;
