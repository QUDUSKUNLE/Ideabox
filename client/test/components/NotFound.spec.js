import React from 'react';
import { shallow } from 'enzyme';
import NotFound from '../../src/components/NotFound';

describe('<NotFound/>', () => {
  it('component should be defined', () => {
    expect(NotFound).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<NotFound />);
    expect(component).toMatchSnapshot();
    expect(component.find('div')).toHaveLength(5);
  });
});
