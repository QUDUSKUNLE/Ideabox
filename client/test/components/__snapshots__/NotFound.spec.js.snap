// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
        >
          <h4>
            IdeaBox
          </h4>
        </div>,
        <div
          className="col s12 flex-center"
        >
          <div
            className="center-align"
          >
            <div
              className="center-align centralize"
            >
              <h3
                className="white-text"
              >
                You\`re lost.
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </h3>
            </div>
          </div>
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            IdeaBox
          </h4>,
          "className": "logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "IdeaBox",
          },
          "ref": null,
          "rendered": "IdeaBox",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="center-align"
          >
            <div
              className="center-align centralize"
            >
              <h3
                className="white-text"
              >
                You\`re lost.
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </h3>
            </div>
          </div>,
          "className": "col s12 flex-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="center-align centralize"
            >
              <h3
                className="white-text"
              >
                You\`re lost.
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </h3>
            </div>,
            "className": "center-align",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3
                className="white-text"
              >
                You\`re lost.
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </h3>,
              "className": "center-align centralize",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "You\`re lost.",
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                ],
                "className": "white-text",
              },
              "ref": null,
              "rendered": Array [
                "You\`re lost.",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
              ],
              "type": "h3",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          >
            <h4>
              IdeaBox
            </h4>
          </div>,
          <div
            className="col s12 flex-center"
          >
            <div
              className="center-align"
            >
              <div
                className="center-align centralize"
              >
                <h3
                  className="white-text"
                >
                  You\`re lost.
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </h3>
              </div>
            </div>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              IdeaBox
            </h4>,
            "className": "logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "IdeaBox",
            },
            "ref": null,
            "rendered": "IdeaBox",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="center-align"
            >
              <div
                className="center-align centralize"
              >
                <h3
                  className="white-text"
                >
                  You\`re lost.
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </h3>
              </div>
            </div>,
            "className": "col s12 flex-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="center-align centralize"
              >
                <h3
                  className="white-text"
                >
                  You\`re lost.
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </h3>
              </div>,
              "className": "center-align",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="white-text"
                >
                  You\`re lost.
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </h3>,
                "className": "center-align centralize",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "You\`re lost.",
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>,
                  ],
                  "className": "white-text",
                },
                "ref": null,
                "rendered": Array [
                  "You\`re lost.",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Home",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                ],
                "type": "h3",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
